### PR TITLE
5739 move logout to footer user

### LINKED ISSUE #5739: Move logout button to Footer -> user
Screenshot 2024-12-11 at 10 18 59 AM

- [ ] Remove logout button from AppDrawer
- [ ] Add popup on click of username in footer, with the logout button
- [ ] Add user info above the logout button

Use footer store select/language select for how to show popup on click of the username.

### PR DESCRIPTION
**Pull request by noel-yeldos**
Fixes #5739 

# 👩🏻‍💻 What does this PR do?
Move logout link from app drawer to a popup in the footer for user. 

Create new component where logout button and user details are housed.

Changes needed as part of GEDSI

Screenshot 2024-12-12 at 3 27 52 PM

## 💌 Any notes for the reviewer?
I had to overwrite some styles for textprops and labelprops to make label and text align. The email now wraps funny if the text is too long. Open to suggestions.

Sidequest to fix: https://github.com/msupply-foundation/open-msupply/pull/5749#discussion_r1881116621

# 🧪 Testing

- [ ] Click on 'admin' (or whatever your username is) in the footer
- [ ] See new popup display 
- [ ] Pop up heading should be user's name
- [ ] User's username and email should be displayed
- [ ] Logout button should be displayed and when clicked should log user out

# 📃 Documentation

- [x] **Part of an epic**: documentation will be completed for the feature as a whole
- [ ] **No documentation required**: no user facing changes or a bug fix which isn't a change in behaviour
- [ ] **These areas should be updated or checked**: 
  1.
  2.

### CODE REVIEW

File: client/packages/host/src/components/Footer/UserDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+                }}
+              />
+              <TextWithLabelRow
+                label={t('label.email')}
+                text={user.email ?? ''}
+                textProps={{ textAlign: 'left', lineHeight: 1.5 }}
+                labelProps={{
+                  sx: { textAlign: 'left', width: 50, lineHeight: 1.5 },
+                }}
+              />
```
Comment (GeronimoJohn) [inline client/packages/host/src/components/Footer/UserDetails.tsx:88]: Regarding comment: `I had to overwrite some styles for textprops and labelprops to make label and text align. The email now wraps funny if the text is too long. Open to suggestions.`

Love that we're trying to solve this! We can try converting this into an [ellipsis](https://developer.mozilla.org/en-US/docs/Web/CSS/text-overflow) with CSS. Just not sure if that's a pattern that we follow @lache-melvin. 

But if we humour the idea of solving it that way. We can utilise the sx prop within `TextWithLabelRow` (not the one on lableProps). And apply the necessary css fields.

Comment (GeronimoJohn) [reply client/packages/host/src/components/Footer/UserDetails.tsx:88]: After looking further with the props. I wonder if we can just pass in `textOverflow: ellipsis` to textProps with a width

Comment (lache-melvin) [reply client/packages/host/src/components/Footer/UserDetails.tsx:88]: Yes we do! Here's an example - important to add the Tooltip if we add the ellipsis so the can see the full email if they need 😁 

https://github.com/msupply-foundation/open-msupply/blob/aeb2842227dd456a9b1f4eaf7288bee648e48cbd/client/packages/common/src/ui/layout/tables/components/Cells/TooltipTextCell/TooltipTextCell.tsx#L11-L15

File: client/packages/host/src/components/Footer/UserDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  useUserDetails,
+  useConfirmationModal,
+  RouteBuilder,
+  PowerIcon,
+  TextWithLabelRow,
+} from '@openmsupply-client/common';
+import { AppRoute } from '@openmsupply-client/config';
+import { PropsWithChildrenOnly } from '@common/types';
+
+export const AdminSelector: FC<PropsWithChildrenOnly> = ({ children }) => {
```
Comment (lache-melvin) [inline client/packages/host/src/components/Footer/UserDetails.tsx:20]: Naming request from me... can we call this `UserDetails`, like the file is called?

Selector makes sense for other popups (selecting a different store or language) but less so here 🙏

File: client/packages/host/src/components/Footer/UserDetails.tsx (hunk 1 above)
Comment (noel-yeldos) [reply client/packages/host/src/components/Footer/UserDetails.tsx]: Thanks @GeronimoJohn and @lache-melvin for your help on this! I've decided to leave the tooltip placement as right as it looks the cleanest.

File: client/packages/host/src/components/Footer/UserDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+      sx={{
+        whiteSpace: 'nowrap',
+        overflowX: 'hidden',
+        overflowY: 'visible',
+        textOverflow: 'ellipsis',
+      }}
+    />
+  );
+
+  return user && Tooltip ? (
```
Comment (GeronimoJohn) [inline client/packages/host/src/components/Footer/UserDetails.tsx:57]: In regards of Tooltip. I'm not quite sure what it is, I've looked into but it seems to be a component? If it is a component, `user` alone should be sufficient 😃

File: client/packages/host/src/components/Footer/UserDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+            </Box>
+          )}
+          {logoutButton}
+        </PaperPopoverSection>
+      }
+    >
+      {children}
+    </PaperClickPopover>
+  ) : (
+    isLoading
```
Comment (GeronimoJohn) [inline client/packages/host/src/components/Footer/UserDetails.tsx:115]: isLoading is just a boolean I think. If the goal is to render an empty item when user is empty. We can just use React's empty fragment `<></>`

**Review by GeronimoJohn — COMMENTED** [review]
Looking real good! Love the tooltip and ellipsis work 🔥. Just left a few comments/questions, mostly me just wondering 😃

File: client/packages/host/src/components/Footer/UserDetails.tsx (hunk 3 above)
Comment (noel-yeldos) [reply client/packages/host/src/components/Footer/UserDetails.tsx:57]: Ah good catch! Just a remnant from me troubleshooting!

**Review by GeronimoJohn — APPROVED** [review]
Looks good to me! 🔥

File: client/packages/host/src/components/Footer/UserDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+                  width: 150,
+                }}
+                labelProps={{
+                  sx: { textAlign: 'left', width: 80, lineHeight: 1.5 },
+                }}
+                showToolTip={true}
+              />
+              <TextWithLabelRow
+                label={t('label.email')}
+                text={user.email ?? ''}
```
Comment (lache-melvin) [inline client/packages/host/src/components/Footer/UserDetails.tsx:91]:
```suggestion
                text={user.email ?? '-'}
```

Bit nicer to communicate no email set :) 

Screenshot 2024-12-16 at 3 08 55 PM

File: client/packages/host/src/components/Footer/UserDetails.tsx [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+    onConfirm: handleLogout,
+    message: t('messages.logout-confirm'),
+    title: t('heading.logout-confirm'),
+  });
+
+  const logoutButton = (
+    <FlatButton
+      startIcon={<PowerIcon fontSize="small" color="primary" />}
+      label={t('logout')}
+      disabled={false}
```
Comment (lache-melvin) [inline client/packages/host/src/components/Footer/UserDetails.tsx:42]: false by default so shouldn't need this at all 😁

**Review by lache-melvin — APPROVED** [review]
couple NITs from me but merge when ready!

File: client/packages/host/src/components/Footer/UserDetails.tsx (hunk 5 above)
Comment (roxy-dao) [reply client/packages/host/src/components/Footer/UserDetails.tsx:91]: global const UNDEFINED_STRING_VALUE 😉

Comment (lache-melvin) [reply client/packages/host/src/components/Footer/UserDetails.tsx:91]: keeping me honest @roxy-dao  thank you 😁

**Comment by mark-prins** [comment]
A nit from me too:

image

* the close button looks nicer with a little room around it - 4px is enough. you could do that rather than the absolute position of `0` (set gap or padding to 1 in the sx)
* all of my users have null first & last names - can we have them show just as empty strings please and not 'null' 🙏 
* the use of `UNDEFINED_STRING_VALUE` as noted above for the email 

thanks! the popup makes the username element more sensible, actually gives it a purpose, and moving the logout is great!
you could go even further and add the job title if you wanted.. but that's confection, I don't think it helps anyone!

File: client/packages/host/src/components/Footer/UserDetails.tsx [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+      }}
+    />
+  );
+
+  return user ? (
+    <PaperClickPopover
+      placement="top"
+      width={300}
+      Content={
+        <PaperPopoverSection label={`${user.firstName} ${user.lastName}`}>
```
Comment (mark-prins) [inline client/packages/host/src/components/Footer/UserDetails.tsx:61]:
```suggestion
        <PaperPopoverSection label={getLocalisedFullName(user.firstName, user.lastName)}>
```

after doing this:
```
  const { getLocalisedFullName } = useIntlUtils();
```

File: client/packages/host/src/components/Footer/Footer.tsx [hunk 8]
```diff
@@ -77,10 +78,14 @@ export const Footer: React.FC = () => {
...
         <>
           <Divider />
-          <PaddedCell>
-            <UserIcon sx={iconStyles} />
-            <Typography sx={textStyles}>{user.name}</Typography>
-          </PaddedCell>
+          <UserDetails>
+            <PaddedCell>
+              <UserIcon sx={iconStyles} />
+              <Tooltip title={`${user.firstName} ${user.lastName}`}>
```
Comment (mark-prins) [inline client/packages/host/src/components/Footer/Footer.tsx:84]: can you please use `getLocalisedFullName` to account for empty strings etc

**Comment by mark-prins** [comment]
@noel-yeldos I've put some suggestions into [this branch](https://github.com/msupply-foundation/open-msupply/compare/5739-move-logout-to-footer-user...5739-move-logout-to-footer-user(suggestion))

which makes the modal look like this:

Screenshot 2024-12-16 at 7 46 39 PM
 
Screenshot 2024-12-16 at 7 54 35 PM


Screenshot 2024-12-16 at 7 45 33 PM

I think the consistent label width is nicer - the other changes are mainly to cope with different data availability 🙏

File: client/packages/host/src/components/Footer/UserDetails.tsx (hunk 5 above)
Comment (noel-yeldos) [reply client/packages/host/src/components/Footer/UserDetails.tsx]: Updated here dbcdd53d553899815728e0849f76c11071c0009d :)

File: client/packages/host/src/components/Footer/UserDetails.tsx (hunk 6 above)
Comment (noel-yeldos) [reply client/packages/host/src/components/Footer/UserDetails.tsx]: Updated here! dbcdd53d553899815728e0849f76c11071c0009d

File: client/packages/host/src/components/Footer/Footer.tsx (hunk 8 above)
Comment (noel-yeldos) [reply client/packages/host/src/components/Footer/Footer.tsx]: Updated here: dbcdd53d553899815728e0849f76c11071c0009d

**Review by lache-melvin — APPROVED** [review]
Looks great Noel - I think let's get this PR in, and can create a new issue for the user info not sticking around when you switch stores 🚀

**Pull request merged**